core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -134,7 +136,8 @@
    private final CacheContainerStats stats;
    private final ConfigurationManager configurationManager;
 
-   @GuardedBy("this")
+   private final ReadWriteLock stoppingLock = new ReentrantReadWriteLock();
+   @GuardedBy("stoppingLock")
    private boolean stopping;
```

Comment:
I looked into this more deeply and prepared this: https://github.com/infinispan/infinispan/pull/4559

@vjuranek @rvansa Please tell me what you think...